### PR TITLE
{vis}[foss/2018a] X11 20180131

### PR DESCRIPTION
**Pull request by RvDijk**
@boegel the components aren't updated, should I look into these manually or is there a script available?

edit: now depends on https://github.com/easybuilders/easybuild-easyblocks/pull/1369

### CODE REVIEW

**Comment by boegel** [comment]
@RvDijk It would be nice to update the extensions, but I'm afraid that's (still) a very manual process...

One more major thing is that you should only add an X11 easyconfig using the `GCCcore` toolchain, so this installation can be used as a dependency for modules installed with `foss` or `intel`.

**Comment by RvDijk** [comment]
@boegel that would require GCCcore versions of Perl as well... So these are coming up! 
I'll also manually check all the extensions.

**Comment by boegel** [comment]
Thanks for the effort @RvDijk!

**Comment by RvDijk** [comment]
@boegel oops closed the wrong one ...

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 6 out of 7 (5 easyconfigs in this PR)
dell.lan - Linux fedora 27, Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz, Python 2.7.14
See https://gist.github.com/5f19888d35e057747d3e649e3ac6af44 for a full test report.

File: easybuild/easyconfigs/x/XML-Parser/XML-Parser-2.44_01-foss-2018a-Perl-5.26.1.eb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+easyblock = 'PerlModule'
+
+name = 'XML-Parser'
+version = '2.44_01'
+versionsuffix = '-Perl-%(perlver)s'
+
+homepage = 'http://search.cpan.org/~toddr/XML-Parser-2.41/'
+description = """This is a Perl extension interface to James Clark's XML parser, expat."""
+
+toolchain = {'name': 'foss', 'version': '2018a'}
```
Comment (boegel) [inline easybuild/easyconfigs/x/XML-Parser/XML-Parser-2.44_01-foss-2018a-Perl-5.26.1.eb]: You can remove this easyconfig, the existing `XML-Parser-2.44_01-GCCcore-6.4.0-Perl-5.26.1.eb` suffices?

File: easybuild/easyconfigs/f/freetype/freetype-2.9-GCCcore-6.4.0.eb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+sources = [SOURCE_TAR_GZ]
+checksums = ['bf380e4d7c4f3b5b1c1a7b2bf3abb967bda5e9ab480d0df656e0e08c5019c5e6']
+
+builddependencies = [
+    ('binutils', '2.28'),
+]
+
+dependencies = [
+    ('bzip2', '1.0.6'),
+    ('libpng', '1.6.32'),
```
Comment (boegel) [inline easybuild/easyconfigs/f/freetype/freetype-2.9-GCCcore-6.4.0.eb]: @RvDijk We should update this to `libpng` 1.6.34 for the `2018a` cycle?

File: easybuild/easyconfigs/x/X11/X11-20180128-GCCcore-6.4.0.eb [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+source_urls = [
+    XORG_LIB_SOURCE,
+    XORG_PROTO_SOURCE,
+    'http://xcb.freedesktop.org/dist/',
+    'http://xkbcommon.org/download/',
+    'http://cgit.freedesktop.org/xorg/util/macros/snapshot',
+    XORG_DATA_SOURCE + '/xkeyboard-config',
+]
+checksums = [
```
Comment (boegel) [inline easybuild/easyconfigs/x/X11/X11-20180128-GCCcore-6.4.0.eb]: This is not being used for the listed components, checksums need to be specified on a per-component basis...

Unfortunately, `--inject-checksums` doesn't support injecting checksums for components yet... :(

File: easybuild/easyconfigs/x/X11/X11-20180128-GCCcore-6.4.0.eb [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    ('libXdmcp', '1.1.2', common_specs),                    # 2015-03-21
+    ('xcb-proto', '1.12', common_specs),                    # 2016-05-18
+    ('libxcb', '1.12', {                                    # 2016-05-18
+        'sources': [SOURCE_TAR_GZ],
+        'start_dir': '%(name)s-%(version)s',
+        'patches': ['libxcb_indent.patch'],
+
+    }),
+    ('xtrans', '1.3.5', common_specs),                      # 2014-09-22
+    ('libxkbcommon', '0.7.2', {                             # 2017-12-16
```
Comment (boegel) [inline easybuild/easyconfigs/x/X11/X11-20180128-GCCcore-6.4.0.eb]: `0.8.0` was released Dec 15th 2017 (0.7.2 dates from Aug'17), it seems like you bumped the date but not the version? :)

**Comment by boegel** [comment]
@RvDijk We should specify the checksums for the X11 components, done in https://github.com/rug-cit-ris/easybuild-easyconfigs/pull/7, but this requires an update to the `Bundle` easyblock, see https://github.com/easybuilders/easybuild-easyblocks/pull/1369 .

**Comment by boegel** [comment]
tests fail because https://github.com/easybuilders/easybuild-easyblocks/pull/1369 isn't merged yet (which also fixes the broken error message), do ignore, I'll retrigger the tests once that PR is merged

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2689.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/eb9bd3a188e200f29b532caea70150e8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2030.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/42ad6b5d6596309f9864710e2b10a4fc for a full test report.

**Comment by boegel** [comment]
Going in, thanks @RvDijk!

**Pull request merged**